From b08b8139ffd35dec77cd1927cec93ec07da9fcf9 Mon Sep 17 00:00:00 2001
From: "Kevin F. Haggerty" <haggertk@lineageos.org>
Date: Fri, 16 Nov 2018 17:39:18 -0700
Subject: [PATCH 11/18] msm8974-common: sepolicy: Label sysfs_msmuart_file,
 resolve denials

* avc: denied { setattr } for name="clock" dev="sysfs" ino=18914
  scontext=u:r:init:s0 tcontext=u:object_r:sysfs_msmuart_file:s0
  tclass=file permissive=1

Change-Id: Iaf5fe6791344dcf419242599eb6c9272c61cd707
---
 sepolicy/common/file_contexts | 1 +
 sepolicy/common/init.te       | 1 +
 2 files changed, 2 insertions(+)

diff --git a/sepolicy/common/file_contexts b/sepolicy/common/file_contexts
index e91a036..7c4bc58 100644
--- a/sepolicy/common/file_contexts
+++ b/sepolicy/common/file_contexts
@@ -31,6 +31,7 @@
 /data/cam_socket3                                       u:object_r:camera_socket:s0
 
 # sysfs
+/sys/devices/[a-f0-9]+\.uart(/.*)?                               u:object_r:sysfs_msmuart_file:s0
 /sys/devices/platform/bcm[0-9]+_bluetooth/rfkill/rfkill0/state   u:object_r:sysfs_bluetooth_writable:s0
 /sys/devices/virtual/camera(/.*)?                                u:object_r:sysfs_camera:s0
 /sys/devices/virtual/input(/.*)?                                 u:object_r:sysfs_input:s0
diff --git a/sepolicy/common/init.te b/sepolicy/common/init.te
index 04028c8..71622f5 100644
--- a/sepolicy/common/init.te
+++ b/sepolicy/common/init.te
@@ -18,6 +18,7 @@ allow init {
     sysfs_input
     sysfs_leds
     sysfs_mdnie
+    sysfs_msmuart_file
     sysfs_sec_bamdmux
     sysfs_sec_barcode_emul
     sysfs_sec_epen
-- 
2.17.1

